<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>658</width>
    <height>568</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ResearchWindow</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>621</width>
     <height>523</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="lineWidth">
        <property name="title">
         <string>Line width:</string>
        </property>
        <widget class="QSpinBox" name="widthSpinBox">
         <property name="geometry">
          <rect>
           <x>200</x>
           <y>50</y>
           <width>42</width>
           <height>22</height>
          </rect>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
        <widget class="QLineEdit" name="widthEditor">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>50</y>
           <width>151</width>
           <height>20</height>
          </rect>
         </property>
         <property name="maxLength">
          <number>2</number>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>120</y>
           <width>305</width>
           <height>400</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>99</width>
           <height>400</height>
          </size>
         </property>
         <property name="title">
          <string>Line color:</string>
         </property>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>101</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Red:</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>90</y>
            <width>91</width>
            <height>16</height>
           </rect>
          </property>
          <property name="text">
           <string>Green:</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>160</y>
            <width>46</width>
            <height>13</height>
           </rect>
          </property>
          <property name="text">
           <string>Blue: </string>
          </property>
         </widget>
         <widget class="QSlider" name="redSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>60</y>
            <width>261</width>
            <height>19</height>
           </rect>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
         </widget>
         <widget class="QSlider" name="greenSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>120</y>
            <width>261</width>
            <height>19</height>
           </rect>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
         </widget>
         <widget class="QSlider" name="blueSlider">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>190</y>
            <width>261</width>
            <height>19</height>
           </rect>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
         </widget>
         <widget class="QPushButton" name="resetButton">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>310</y>
            <width>261</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
         <widget class="QPushButton" name="exitButton">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>340</y>
            <width>261</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Exit</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="MyFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <property name="lineWidth">
         <number>2</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyFrame</class>
   <extends>QFrame</extends>
   <header>myframe.h</header>
   <container>1</container>
   <slots>
    <slot>getPen(QPen)</slot>
    <slot>getWidth(int)</slot>
    <slot>getReset()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>widthEditor</sender>
   <signal>textEdited(QString)</signal>
   <receiver>Widget</receiver>
   <slot>getString(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>636</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Widget</receiver>
   <slot>getInt(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Widget</sender>
   <signal>sendString(QString)</signal>
   <receiver>widthEditor</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>640</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Widget</sender>
   <signal>sendInt(int)</signal>
   <receiver>widthSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>643</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>Widget</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>187</x>
     <y>546</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Widget</receiver>
   <slot>redefineValues()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>643</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Widget</receiver>
   <slot>redefineValues()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>646</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Widget</receiver>
   <slot>redefineValues()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>645</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Widget</sender>
   <signal>sendPen(QPen)</signal>
   <receiver>frame</receiver>
   <slot>getPen(QPen)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>590</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Widget</receiver>
   <slot>getWidth(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>643</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Widget</sender>
   <signal>sendWidth(int)</signal>
   <receiver>frame</receiver>
   <slot>getWidth(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>582</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>Widget</receiver>
   <slot>getReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>452</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Widget</sender>
   <signal>sendReset()</signal>
   <receiver>frame</receiver>
   <slot>getReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>542</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>559</x>
     <y>498</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>sendString(QString)</signal>
  <signal>sendInt(int)</signal>
  <signal>sendPen(QPen)</signal>
  <signal>sendWidth(int)</signal>
  <signal>sendReset()</signal>
  <slot>getInt(int)</slot>
  <slot>getString(QString)</slot>
  <slot>redefineValues()</slot>
  <slot>getWidth(int)</slot>
  <slot>getReset()</slot>
 </slots>
</ui>
